packages/newsroom-manager/src/SignConstitution.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from "react";
+import { connect, DispatchProp } from "react-redux";
+// import { addGovernmentData } from "./actionCreators";
+// import { getConstitutionHash, getConstitutionUri, getNewsroom, signMessage } from "../../apis/civilTCR";
+import {
+  StepHeader,
+  StepProps,
+  StepStyled,
+  Collapsable,
+  // SignConstitutionButton,
+  StepDescription,
+} from "@joincivil/components";
+// import styled from "styled-components";
+import { Map } from "immutable";
+// import { getCivil } from "../../helpers/civilInstance";
+
+// const StyledLegalIframe = styled.iframe`
+// border-width: 1px;
+// height: 15rem;
+// margin: 0 0 2rem;
+// width: 100%;
+// `;
+
+export interface SignConstitutionProps {
+  address?: string;
+}
+
+export interface SignConstitutionReduxProps {
+  government?: Map<string, string>;
+  user?: any;
+}
+
+export interface SignConstitutionState {
+  isNewsroomOwner?: boolean;
+}
+
+class SignConstitutionComponent extends React.Component<
+  DispatchProp<any> & StepProps & SignConstitutionProps & SignConstitutionReduxProps,
+  SignConstitutionState
+> {
+  constructor(props: StepProps & SignConstitutionProps & SignConstitutionReduxProps) {
+    super(props);
+    this.state = {
+      isNewsroomOwner: false,
+    };
+  }
+
+  public async componentDidMount(): Promise<void> {
+    if (this.props.address && this.props.address.length && !this.props.disabled) {
+      // const isOwner = await this.isNewsroomOwner();
+      // this.setState({ isNewsroomOwner: isOwner });
+    }
+    // return this.initGovernmentData();
+  }
+
+  public async componentDidUpdate(
+    prevProps: StepProps & SignConstitutionProps & SignConstitutionReduxProps,
+    prevState: SignConstitutionState,
+  ): Promise<void> {
+    // const prevNewsroomAddress = prevProps.address;
+    // const newsroomAddress = this.props.address;
+    // const newState: SignConstitutionState = {};
+    // const isUpdatedAddress = !prevNewsroomAddress && newsroomAddress && newsroomAddress.length;
+    // const isUpdatedUser = prevProps.user !== this.props.user;
+    // if (isUpdatedAddress || (isUpdatedUser && !this.props.disabled)) {
+    // const isOwner = await this.isNewsroomOwner();
+    // newState.isNewsroomOwner = isOwner;
+    // }
+    // if (Object.keys(newState).length) {
+    // this.setState(newState);
+    // }
+  }
+
+  public render(): JSX.Element {
+    // const civil = getCivil();
+    return (
+      <StepStyled disabled={this.props.disabled} index={this.props.index || 0}>
```

Comment:
Just to verify, the "coming soon" steps haven't been converted to use `CivilContext` and work within the Newsroom package structure b/c they're coming soon?

Reply:
Yeah, I didn't want to move all the civil api stuff in here last night.  So i commented it out to come back to it after launch